src/Agent.Worker/Worker.cs

```diff
@@ -109,13 +111,22 @@ private void InitializeSecretMasker(Pipelines.AgentJobRequestMessage message)
             Trace.Entering();
             ArgUtil.NotNull(message, nameof(message));
             ArgUtil.NotNull(message.Resources, nameof(message.Resources));
-
             // Add mask hints for secret variables
             foreach (var variable in (message.Variables ?? new Dictionary<string, VariableValue>()))
             {
                 if (variable.Value.IsSecret)
                 {
                     HostContext.SecretMasker.AddValue(variable.Value.Value);
+                    // for variables, it is possible that they are used inside a shell which would strip off surrounding quotes
+                    // so, if the value is surrounded by quotes, add a quote-timmed version of the secret to our masker as well
+                    // This addresses issue #2525
+                    foreach (var quoteChar in _quoteLikeChars)
+                    {
+                        if (variable.Value.Value.StartsWith(quoteChar) && variable.Value.Value.EndsWith(quoteChar))
```

Comment:
I would probably do `(variable.Value.Value.StartsWith(quoteChar) || variable.Value.Value.EndsWith(quoteChar))`. I can't quite pin down a case where that would matter, but its very cheap and won't cause undesirable masking.

Reply:
I did `&&` because I explicitly only wanted to trim in the case where it begins and ends with the quote char.
If their secret is "foo , I am not sure we want to trim it.

Reply:
I guess my point is that its almost certainly fine - its a slight overmasking, but probably a good one at that point. I'm fine leaving it, I guess the only case I'm really worried about is someone fat-fingering a secret (e.g. tries to type `"foo"` but types `"foo` or even `"foo'`). Not too worried about it either way though, I'll leave that up to you.